src/main/java/com/marklogic/kafka/connect/sink/MarkLogicSinkTask.java

```diff
@@ -114,7 +114,11 @@ public void put(final Collection<SinkRecord> records) {
 			if (logger.isDebugEnabled()) {
 				logger.debug("Processing record value {} in topic {}", record.value(), record.topic());
 			}
-			writeBatcher.add(sinkRecordConverter.convert(record));
+			if (record.value() != null) {
```

Comment:
Can the record be null as well? i.e. can the Collection<SinkRecord> contain null objects in it? Seems unlikely, but easy enough to check - and that check should go before the debug logging above.